DVCodingStyle.md

```diff
@@ -0,0 +1,864 @@
+    `gtn` shorthand macros used in place of `get_full_name()` and `get_type_name()`.
+    This will allow for easier debugging should something go wrong.
+7.  The `UVM_NONE` and `UVM_FULL` verbosity levels must not be used for any
+    report messages.
+    Recommended criteria for the choice of `VERBOSITY` value for `uvm_info`
+    messages:
+    *   `UVM_LOW` - important messages that occur a small number of times
+        during a simulation.
+        *   Instantiation of testbench components.
+        *   Reset assert/deassert, start/end of DUT initialization.
+        *   Start/end of traffic generation.
+    *   `UVM_MEDIUM` - slightly more detailed messages.
+        *   Contents of CSR fields during CSR transactions.
+        *   Reporting of significant events inside the DUT - fifo full,
+        fifo empty, and so on.
+    *   `UVM_HIGH` - Any information about DUT activity, reference model,
+        and testbench that could be helpful for monitoring stimulation progress.
+        *   Parsing/interpretation of fields in requests to the DUT and reference model.
+        *   Read/write operations to a memory model.
+        *   Status of transactions over SystemVerilog interfaces.
+        *   Values of data output by the DUT.
+    *   `UVM_DEBUG` - Extremely specific information used for debugging
+        failures.
+        *   Any CSR exclusions.
+        *   Interface buses used to drive the interface protocol.
+
+
+### DPI and C Connections
+
+1.  DPI and native SystemVerilog calls have the same semantics and
+    are thus indistinguishable.
+    To preserve readability, a decorator must be added as such:
+    *   Functions/tasks exported from SV to C must be prefixed with `sv_dpi_`.
+
+        ```systemverilog
+        export "DPI" sv_dpi_adder = adder_function;
+        ```
+
+    *   Functions imported from C to SV must be prefixed with `c_dpi_`.
+
+        ```systemverilog
+        extern int c_dpi_adder(const int a, const int b);
+        ```
+
+2.  DPI usage is recommended in the following scenarios:
+    *   Interfacing with C/C++ based reference models.
+    *   Usage of optimized C/C++ libraries.
+    *   Emulation.
+3. DPI usage is not recommended when the functionality can be handled natively
+    in SV.
+4.  General DPI coding guidelines
+    *   Do not cross the language boundary frequently.
+    *   If a SV->C call consumes time and the C routine is not thread safe,
+    protect the call with a semaphore to avoid synchronization issues.
+    *   In general, pass only basic data types as arguments to DPI calls.
+
+
+## SystemVerilog Language Features
+
+### Function Declarations
+
+Functions declared within packages, as well as any other static entities such as
+modules and interfaces, must be explicitly scoped with either the `static` or
+`automatic` keyword.
+
+### Randomization
+
+Good constraint coding style is important for faster simulations and better
+simulation performance.
```

Comment:
I would suggest we recommend using a constraint name that matches what is being constrained.
 for the example below I would suggest a naming scheme with either

a_con
con_a
or 
c_a

Reply:
That is a good point you bring up, the style that we have generally used so far is `<constraint_name>_c`, let us standardize that format if that sounds good to you, and I'll add it to the style guide?
@sriyerg @cindychip @weicaiyang what do you think?

Reply:
Sure I agree with adding `<constraint_name>_c`

Reply:
agree with <constraint_name>_c, too

Reply:
I am ok with _c also